### ISSUE TITLE
Peer dependencies too strict?

### ISSUE THREAD

**Issue by wojtekmaj** [WestpacGEL/GEL#774]
**Describe the bug**

I have noticed that in #719 `react` and `react-dom` in peerDependencies were set to `^17.0.2`. If I'm not mistaken, GEL seems to work perfectly fine with 17.0.0, and even 16.8.0. 

I can imagine some teams lagging behind slightly, as many packages were not yet updated to declare React 17 compatibility. They would be unable to install GEL because of peerDependencies being too strict, especially considering npm v7 (ridiculous!) handling of peerDependencies.

I would reconsider the definition of peerDependencies and perhaps use `^16.8.0 || ^17.0.0` instead.

You can use [Matrix jobs](https://circleci.com/blog/circleci-matrix-jobs/) to run tests on multiple versions of React to ensure compatibility.

**Comment by dominikwilkowski** [comment]
yeah good point. We just never got around to do it. Should be `"react": "^16 || ^17"` hey
